ibtk/src/lagrangian/FEProjector.cpp

```diff
@@ -742,6 +741,7 @@ FEProjector::computeL2Projection(PetscVector<double>& U_vec,
         if (qrule_is_nodal(fe_type, qrule.get()))
         {
             // use the lumped matrix as the preconditioner:
+            PetscMatrix<double>& lumped_mass = *buildLumpedL2ProjectionSolver(system_name).second;
```

Comment:
@AminKolah, I think this is the change that is causing tests to fail.

Reply:
The change that is made here is basically saying that only build the lumped L2 projection solver if the qrule is nodal. Is that not correct? There is a `qrule_is_nodal`  assertion in `buildLumpedL2ProjectionSolver` itself that triggers an error if qrule is not nodal.

Reply:
I suspect the reason is more banal - we rely on log files to verify that results haven't changed, so moving this down changes the order in which lines are printed (and causes a test failure). Is there a reason this was moved?

Reply:
So currently the 3 choices with USE_DISCON_ELEMS = TRUE are 1) CONSTANT MONOMIAL, 2) FIRST order L2_LAGRANGE and 3) FIRST order MONOMIAL. That line was moved down inside the if condition because currently the 3rd option (FIRST order MONOMIAL) fails the assertion for `qrule_is_nodal`. Practically speaking, we may never even need to use FIRST order MONOMIAL because the second option (FIRST order L2_LAGRANGE) would give the same result. So I can put that line back if we all agree there is no need to use FIRST order MONOMIAL for the jump discontinuous systems.

Reply:
To be more precise, this is the line that causes an error when the first order monomial is used. @drwells is it necessary to keep this check?
https://github.com/IBAMR/IBAMR/blob/833bf42f40123190d5fe6afe62f5b26f1f9117b2/ibtk/src/lagrangian/FEProjector.cpp#L127

Reply:
If I recall correctly, the monomial space is `{1, x, y}` etc: past using just constants, there is no corresponding nodal quadrature rule, so the check is correct. What do you expect to happen here? I think `L2_Lagrange` is going to be the right choice.